### PR TITLE
[fix] Skipped url should be really ignored.

### PR DESCRIPTION
**Pull request by opi**

### CODE REVIEW

**Comment by alexAubin** [comment]
Do you know what's the status of this ? It might fix https://dev.yunohost.org/issues/717

File: access.lua [hunk 1]
```diff
@@ -227,25 +227,30 @@ end
...
-        or  hlp.string.starts(ngx.var.uri..hlp.uri_args_string(), url))
-        and not is_protected() then
-            return hlp.pass()
+function is_skipped()
+    if conf["skipped_urls"] then
+        for _, url in ipairs(conf["skipped_urls"]) do
+            if (hlp.string.starts(ngx.var.host..ngx.var.uri..hlp.uri_args_string(), url)
+            or  hlp.string.starts(ngx.var.uri..hlp.uri_args_string(), url))
+            and not is_protected() then
+                hlp.pass()
```
Comment (opi) [inline access.lua:236]: this should not be a part of this function. `is_skipped` must only return true or false

File: access.lua [hunk 2]
```diff
@@ -227,25 +227,30 @@ end
...
-        if (string.match(ngx.var.host..ngx.var.uri..hlp.uri_args_string(), regex)
-        or  string.match(ngx.var.uri..hlp.uri_args_string(), regex))
-        and not is_protected() then
-            return hlp.pass()
+    if conf["skipped_regex"] then
+        for _, regex in ipairs(conf["skipped_regex"]) do
+            if (string.match(ngx.var.host..ngx.var.uri..hlp.uri_args_string(), regex)
+            or  string.match(ngx.var.uri..hlp.uri_args_string(), regex))
+            and not is_protected() then
+                hlp.pass()
```
Comment (opi) [inline access.lua:247]: same as before

File: access.lua [hunk 3]
```diff
@@ -344,14 +349,15 @@ if auth_header then
...
 --
 
-hlp.flash("info", hlp.t("please_login"))
-local back_url = ngx.var.scheme .. "://" .. ngx.var.host .. ngx.var.uri .. hlp.uri_args_string()
-return hlp.redirect(conf.portal_url.."?r="..ngx.encode_base64(back_url))
+if not is_skipped() then
+    hlp.flash("info", hlp.t("please_login"))
+    local back_url = ngx.var.scheme .. "://" .. ngx.var.host .. ngx.var.uri .. hlp.uri_args_string()
+    return hlp.redirect(conf.portal_url.."?r="..ngx.encode_base64(back_url))
+end
```
Comment (opi) [inline access.lua:363]: `else hlp.pass()`

**Comment by alexAubin** [comment]
[Semi-automatic message] 
Trying to clean PR which are opened and inactive since more than 6 months ~ 1 year ... I propose to close this PR and keep it in the "Horizon" milestone in case somebody wants to take care of it later. But in the meantime there's no point in keeping all of these opened if there's no indication that someone might work on this soon.

**Pull request closed without merging**